packages/opentelemetry-plugin-http/src/utils.ts

```diff
@@ -217,6 +217,46 @@ export const setSpanWithError = (
   span.setStatus(status);
 };
 
+/**
+ * Adds attributes for content-length and content-encoding HTTP headers
+ * @param { OutgoingHttpHeaders | IncomingHttpHeaders } headers http headers
+ * @param { Attributes } Attributes span attributes
+ * @param { boolean } isRequest set true for setting request content-header
+ */
+export const setContentLengthAttributes = (
+  headers: OutgoingHttpHeaders | IncomingHttpHeaders,
+  attributes: Attributes,
+  isRequest: boolean
+) => {
+  let isCompressed = false;
+
+  if (headers['content-length'] === undefined) return;
+
+  if (
+    headers['content-encoding'] &&
+    headers['content-encoding'] !== 'identity'
```

Comment:
According to this
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding
the 
"Content-Encoding" can have multiple values

should you rather then compare like this ?
```
headers['content-encoding'].indexOf('identity') >= 0
```

Reply:
Maybe taking advantage of this API would be a good idea since it is case insensitive? https://nodejs.org/api/http.html#http_request_getheader_name

Reply:
We're actually verifying that the content-encoding is _not_ identity. This was a check to see if the content is compressed. It seems like the only way that `content-encoding` can have a value and the content be uncompressed is if its value is `identity`, which we are checking _against_. Perhaps it is possible to have `identity, identity`? In which case this check would fail to mark it as uncompressed, but is that possible?

Reply:
I think a simple string check is probably fine in this case. Situation where there are multiple encodings but they are all `'identity'` seems exceedingly uncommon. The headers are also case insensitive, so you will miss `'Content-Encoding'`.

```typescript
function isCompressed(message: IncomingMessage): boolean {
  // case insensitive lookup
  const encoding = message.getHeader('content-encoding');
  // check safe against `any` return type from `getHeader`
  return typeof encoding === 'string' && encoding.toLowerCase() !== 'identity';
}
```